### ISSUE TITLE
Support attribute on tag helper string properties for selections.

### ISSUE THREAD

**Issue by rjamesnw** [aspnet/Mvc#7426]
It seems only enum support for selections exists for Tag Helpers.  There are cases with custom tags where I'd like to give some selections but NOT limit to them.  I think it would be a great idea to have an attribute  - i.e. [HtmlAttributeSelections("1", "2", "3", etc...)] for strings, or [HtmlAttributeSelections(1, 2, 3, etc...)] for numbers, and so on;  Not sure how "smart" the tooling can get there, but just a thought. ;)  I looked for other issues on this but I didn't find any.

**Comment by mkArtakMSFT** [comment]
Hi @rjamesnw. Can you please provide more details regarding how this is going to be used?

**Comment by mkArtakMSFT** [comment]
@DamianEdwards FYI

**Comment by DamianEdwards** [comment]
This is an interesting idea and I can see the utility, but the implementation would be interesting. When you're typing in attribute values you're actually typing C# assignment expression values (i.e. you're literally typing the value being assigned to a C# property), and the IntelliSense provided there is all coming from the C# editor. We'd have to add additional values to the statement completion list that is shown. I think that capability was added to the editor in an update in 15.4 or 15.5, so it might be possible today, but not sure how much work.

Suggestions like this one that require changes to the Razor tooling in addition with the runtime are typically always longer lead and triaged with a lot more scrutiny, just due to the number of moving parts involved. Just setting some expectations.

So, while I like the idea, I think it currently falls into the category of things that will be unlikely to be invested in for quite a while.

**Comment by rjamesnw** [comment]
At lease you like the idea, it’s a start! lol ;) Can you think of a workaround where I can use an enum OR string for the same attribute?

**Comment by DamianEdwards** [comment]
No, we have an issue on that already (attribute overloading) but it's also on the backlog.

**Comment by rjamesnw** [comment]
@mkArtakMSFT I was creating special input tag helpers (as a library) and I wanted to create a global “<xt-input>” type element with “type” that has all the same “<input>” types, but I didn’t like the idea of hard-coding attributes to an enum only. I was also thinking about “suggested” common values, such as button labels “Ok”, “Cancel”, “Yes”, “No”, etc, but keep it text entry so users can enter a different button name.
